etcd-manager/pkg/volumes/scaleway/discovery.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scaleway
+
+import (
+	"fmt"
+
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/etcdadm/etcd-manager/pkg/privateapi/discovery"
+)
+
+// ScwVolumes also allows the discovery of peer nodes
+var _ discovery.Interface = &ScwVolumes{}
+
+// Poll returns all etcd cluster peers.
+func (a *ScwVolumes) Poll() (map[string]discovery.Node, error) {
+	peers := make(map[string]discovery.Node)
+
+	klog.V(2).Infof("Discovering volumes for %q", a.nameTag)
+	etcdVolumes, err := getMatchingVolumes(a.instanceAPI, a.zone, a.matchTags)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get matching volumes: %w", err)
+	}
+
+	for _, volume := range etcdVolumes {
+		if volume.Server == nil {
+			// Volume doesn't have a server attached yet
+			continue
+		}
+		serverID := volume.Server.ID
+
+		server, err := a.instanceAPI.GetServer(&instance.GetServerRequest{
+			ServerID: serverID,
+			Zone:     a.zone,
+		})
+		if err != nil || server == nil {
+			return nil, fmt.Errorf("failed to get the running server: %s", err)
```

Comment:
Nit: when wrapping errors (which is a good idea), we should use `: %w` now (in the past %v was recommended, but %w preserves the "inner" error in a form that can be inspected from code)